WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -125,18 +132,54 @@ class PostListFragment : Fragment() {
                 postListAdapter.updateItemLayoutType(layoutType)
             }
         })
-        viewModel = ViewModelProviders.of(this, viewModelFactory)
-                .get<PostListViewModel>(PostListViewModel::class.java)
-        viewModel.start(mainViewModel.getPostListViewModelConnector(authorFilter, postListType))
-        viewModel.pagedListData.observe(this, Observer {
-            it?.let { pagedListData -> updatePagedListData(pagedListData) }
-        })
+
+        val authorFilter: AuthorFilterSelection = requireNotNull(arguments)
+                .getSerializable(EXTRA_POST_LIST_AUTHOR_FILTER) as AuthorFilterSelection
+        val postListViewModelConnector = mainViewModel.getPostListViewModelConnector(authorFilter, postListType)
+
+        viewModel = ViewModelProviders.of(this, viewModelFactory).get<PostListViewModel>(PostListViewModel::class.java)
+        viewModel.start(postListViewModelConnector)
+
+        initObservers()
+    }
+
+    private fun initObservers() {
+        if (postListType == SEARCH) {
+            mainViewModel.searchQuery.observe(this, Observer {
+                if (TextUtils.isEmpty(it)) {
+                    postListAdapter.submitList(null)
+                }
+
+                searchHandler.removeCallbacksAndMessages(null)
+                searchHandler.postDelayed({
+                    viewModel.search(it)
+                }, SEARCH_DELAY_MS)
+            })
+        }
+
         viewModel.emptyViewState.observe(this, Observer {
             it?.let { emptyViewState -> updateEmptyViewForState(emptyViewState) }
         })
+
         viewModel.isFetchingFirstPage.observe(this, Observer {
-            swipeRefreshLayout?.isRefreshing = it == true
+            if (postListType != SEARCH) {
+                swipeRefreshLayout?.isRefreshing = it == true
+            } else {
+                // most of the time search is pretty fast, so we don't need to show any progress indication
+                // we delay progress indicator in case request takes longer then expected
+                if (it == true) {
```

Comment:
Honestly, I'm not really sure how to do this. My first idea was to implement a delay like this:

```
 _isFetchingFirstPage.addSource(pagedListWrapper.isFetchingFirstPage) {
                searchProgressJob?.cancel()
                searchProgressJob = launch {
                    delay(SEARCH_PROGRESS_INDICATOR_DELAY_MS)
                    searchProgressJob = null
                    if (isActive) {
                        _isFetchingFirstPage.value = it
                    }
                }
            }
```

But maybe because `isFetchingFirstPage` is also tied to empty view, this messes up the list and I end up getting blank screens sometimes.

Reply:
Ok, looks like with other changes +some minor modifications it works ok :)